### PR TITLE
Basic migration test from 1.5 to 2.0

### PR DESCRIPTION
**Pull request by shino**
This PR adds "Riak CS and Riak are in the same box" case of [1].

Rough description of test steps:
1. Setup system with previous versions for riak/stanchion/cs
2. Upgrade stanchion
3. Upgrade node 1 of riak and CS
4. Upgrade node 2 of riak and CS

Between each steps, create user, create buckets, PUT
Objects (including Multipart and PUT Copy), List Objects, DELETE
Objects and Delete buckets.  Also access stats flush, storage
calculation and GC are done for each step.  After all steps are
completed, Delete all remaining objects and buckets and GC them, then
merge bitcask files and assert data files are small enough.

[1] https://github.com/basho/riak_cs/blob/develop/RELEASE-NOTES.md#riak-cs-and-riak-are-in-the-same-box

### CODE REVIEW

**Comment by kuenishi** [comment]
I'd recommend rebasing up to latest develop that generates 2.0.0-xxx version name so as to match version regexp.

**Comment by shino** [comment]
rebased & f-pushed.

**Comment by shino** [comment]
Rebased and force pushed.
Ready for review :cherry_blossom:

File: riak_test/src/cs_suites.erl [hunk 1]
```diff
@@ -0,0 +1,574 @@
...
+obj_bytes_actual(B, K)                       -> kb(byte_size(lead(B, K))).
+
+kb(KB) -> KB * 1024.
+
+mb(MB) -> MB * 1024 * 1024.
+
+merge_all_bitcask(#state{riak_nodes=Nodes, riak_vsn=Vsn} = _State) ->
+    [trigger_bitcask_merge(N, Vsn) || N <- Nodes],
+    %% Wait for merge worker finished
+    [rt:wait_until(
```
Comment (kuenishi) [inline riak_test/src/cs_suites.erl]: You have to verify the result of `rt_wait:until/2` before going next. In my classic old **HDD** env this got timed out and extending timeout made it success. I'd also recommend setting special retry delay here, like twice bigger than existing value, by using `rt_wait:until/3`.

File: riak_test/src/cs_suites.erl [hunk 2]
```diff
@@ -0,0 +1,574 @@
...
+       N,
+       fun(Node) ->
+               Status = rpc:call(Node, bitcask_merge_worker, status, []),
+               lager:debug("Wait util bitcask_merge_worker finished on ~p:"
+                           " status=~p~n", [Node, Status]),
+               Status =:= {0, undefined}
+       end) || N <- Nodes],
+    %% Wait for merge delete worker finished
+    [rt:wait_until(
```
Comment (kuenishi) [inline riak_test/src/cs_suites.erl]: ditto. Actually this line failed for me.

File: riak_test/src/cs_suites.erl [hunk 3]
```diff
@@ -0,0 +1,574 @@
...
+mb(MB) -> MB * 1024 * 1024.
+
+merge_all_bitcask(#state{riak_nodes=Nodes, riak_vsn=Vsn} = _State) ->
+    [trigger_bitcask_merge(N, Vsn) || N <- Nodes],
+    %% Wait for merge worker finished
+    [rt:wait_until(
+       N,
+       fun(Node) ->
+               Status = rpc:call(Node, bitcask_merge_worker, status, []),
+               lager:debug("Wait util bitcask_merge_worker finished on ~p:"
```
Comment (kuenishi) [inline riak_test/src/cs_suites.erl]: I'd recommend updating English as I pointed out in hipchat.

File: riak_test/src/cs_suites.erl [hunk 4]
```diff
@@ -0,0 +1,574 @@
...
+                           " status=~p~n", [Node, Status]),
+               Status =:= {0, undefined}
+       end) || N <- Nodes],
+    %% Wait for merge delete worker finished
+    [rt:wait_until(
+       N,
+       fun(Node) ->
+               QL = rpc:call(Node, bitcask_merge_delete, queue_length, []),
+               lager:debug("Wait util bitcask_merge_delete finished on ~p:"
```
Comment (kuenishi) [inline riak_test/src/cs_suites.erl]: ditto.

File: riak_test/src/cs_suites.erl (hunk 1 above)
Comment (kuenishi) [inline riak_test/src/cs_suites.erl]: My fail setting was 

``` erlang
{default, [
    {rt_max_wait_time, 260000},
    {rt_retry_delay, 5000}
]}.
```

That successed.
Failed setting was for 180000 and 1000.

**Comment by shino** [comment]
Updated, ready for review again.

**Comment by shino** [comment]
pushed bug fix

File: riak_test/src/cs_suites.erl [hunk 5]
```diff
@@ -0,0 +1,581 @@
...
+         abs -> filename:join([BitcaskAbsRoot, VnodeName, F])
+     end ||
+        F <- Fs0, filename:extension(F) =:= ".data"].
+
+%% Assert bitcask data is "small"
+%% 1) The number of *.data files should be =< 2
+%% 2) Each *.data file size should be < 32 KiB
+check_data_files_are_small_enough(Node, Vsn) ->
+    VnodeNames = bitcask_vnode_names(Node, Vsn),
+    check_data_files_are_small_enough(Node, Vsn, VnodeNames).
```
Comment (kuenishi) [inline riak_test/src/cs_suites.erl:531]: I would prefer `lists:all/2` here because it's clearer (to me).

Comment (shino) [inline riak_test/src/cs_suites.erl:531]: `lists:all(fun(VNodeName) -> check_data_files_are_small_enough(Node, Vsn, VnodeName) end, VNodeNames).`

Hmm.. not so clearer for me :see_no_evil: The refactoring is left as an exercise for interested persons :sunglasses:

**Comment by shino** [comment]
@borshop merge

**Pull request merged**